Source/common/new.santa.proto

```diff
@@ -0,0 +1,283 @@
+syntax = "proto3";
+
+import "google/protobuf/any.proto";
+
+option objc_class_prefix = "SNTPB";
+
+package santa;
+
+message UserInfo {
+  int32 uid = 1;
+  string name = 2;
+}
+
+message GroupInfo {
+  int32 gid = 1;
+  string name = 2;
+}
+
+message ProcessID {
+  int32 pid = 1;
+  int32 pidversion = 2;
+}
+
+message NanoTime {
+    int64 sec;
+    int64 nsec;
+}
+
+message MicroTime {
+    int64 sec;
+    int64 usec;
+}
+
+message CodeSignature {
+  bytes cdhash = 1;
+  string signing_id = 2;
+  string team_id = 3;
+}
+
+message Stat {
+  int32 dev = 1;
+  uint32 mode = 2;
+  uint32 nlink = 3;
+  uint64 ino = 4;
+  UserInfo user = 5;
+  GroupInfo group = 6;
+  int32 rdev = 7;
+  NanoTime a_time = 8;
+  NanoTime m_time = 9;
+  NanoTime c_time = 10;
+  NanoTime b_time = 11;
+  int64 size = 12;
+  int64 blocks = 13;
+  int32 blksize = 14;
+  uint32 flags = 15;
+  int32 gen = 16;
+}
+
+message Hash {
+  enum HashAlgo {
+    HASH_ALGO_UNKNOWN = 0;
+    HASH_ALGO_MD5 = 1;
+    HASH_ALGO_SHA1 = 2;
+    HASH_ALGO_SHA256 = 3;
+  }
+
+  HashAlgo type = 1;
+  string hash = 2;
+}
+
+message File {
+  string path = 1;
+  bool truncated = 2;
+  Stat stat = 3;
+  Hash hash = 4;
+}
+
+message FileDescriptor {
+  int32 fd = 1;
+  uint32 type = 2;
+  uint64 pipe_id = 3;
+}
+
+message ProcessInfo {
+  ProcessID id = 1;
+  ProcessID parent_id = 2;
+  ProcessID responsible_id = 3;
+
+  int32 original_id = 4;
+  int32 group_id = 5;
+  int32 session_id = 6;
+  UserInfo effective_user = 7;
+  GroupInfo effective_group = 8;
+  UserInfo real_user = 9;
+  GroupInfo real_group = 10;
+
+  bool is_platform = 11;
+  bool is_es_client = 12;
+  CodeSignature code_signature = 13;
+  uint32 cs_flags = 14;
+
+  File executable = 15;
+  File tty = 16;
+
+  MicroTime start_time = 17;
+}
+
+message CertificateInfo {
+  Hash hash = 1;
+  string common_name = 2;
+}
+
+message Execution {
+  enum Decision {
```

Comment:
nit: I generally find it easier to comprehend embedded enums when they're directly above the field that uses them.